go/arrow/avro/common.go

```diff
@@ -0,0 +1,162 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package avro
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/apache/arrow/go/v13/arrow"
+	"github.com/apache/arrow/go/v13/arrow/memory"
+)
+
+var (
+	ErrMismatchFields = errors.New("arrow/avro: number of records mismatch")
+)
+
+// Option configures an Avro reader/writer.
+type Option func(config)
+type config interface{}
+
+// WithTopLevel specifies whether to include an Avro schema's top level record
+// or only its fields.
+func WithTopLevel(b bool) Option {
+	return func(cfg config) {
+		switch cfg := cfg.(type) {
+		case *OCFReader:
+			cfg.topLevel = b
+		default:
+			panic(fmt.Errorf("arrow/avro: unknown config type %T", cfg))
+		}
+	}
+}
+
+// WithAllocator specifies the Arrow memory allocator used while building records.
+func WithAllocator(mem memory.Allocator) Option {
+	return func(cfg config) {
+		switch cfg := cfg.(type) {
+		case *OCFReader:
+			cfg.mem = mem
+		default:
+			panic(fmt.Errorf("arrow/avro: unknown config type %T", cfg))
+		}
+	}
+}
+
+// WithChunk specifies the chunk size used while reading Avro OCF files.
+//
+// If n is zero or 1, no chunking will take place and the reader will create
+// one record per row.
+// If n is greater than 1, chunks of n rows will be read.
+// If n is negative, the reader will load the whole Avro OCF file into memory and
+// create one big record with all the rows.
+func WithChunk(n int) Option {
+	return func(cfg config) {
+		switch cfg := cfg.(type) {
+		case *OCFReader:
+			cfg.chunk = n
+		default:
+			panic(fmt.Errorf("arrow/avro: unknown config type %T", cfg))
+		}
+	}
+}
+
+// DefaultNullValues is the set of values considered as NULL values by default
+// when Reader is configured to handle NULL values.
+var DefaultNullValues = []string{"", "NULL", "null"}
+
+// WithNullReader sets options for a CSV Reader pertaining to NULL value
+// handling. If stringsCanBeNull is true, then a string that matches one of the
+// nullValues set will be interpreted as NULL. Numeric columns will be checked
+// for nulls in all cases. If no nullValues arguments are passed in, the
+// defaults set in NewReader() will be kept.
+//
+// When no NULL values is given, the default set is taken from DefaultNullValues.
+func WithNullReader(stringsCanBeNull bool, nullValues ...string) Option {
+	return func(cfg config) {
+		switch cfg := cfg.(type) {
+		case *OCFReader:
+			cfg.stringsCanBeNull = stringsCanBeNull
+
+			if len(nullValues) == 0 {
+				nullValues = DefaultNullValues
+			}
+			cfg.nulls = make([]string, len(nullValues))
+			copy(cfg.nulls, nullValues)
+		default:
+			panic(fmt.Errorf("arrow/avro: unknown config type %T", cfg))
+		}
+	}
+}
```

Comment:
see previous comment. doesn't avro have a concept of a null value already so we don't need to have the "null reader" stuff that you need to do for a cSV or other text based reader. (For example, note that there's nothing like this in the JSON reader because if it needs to be null, it should be a JSON null)

Reply:
removed in the latest commit